test/controllers/ContactDetailsControllerSpec.scala

```diff
@@ -16,7 +16,7 @@
 
 package controllers
 
-import controllers.actions.DataRetrievalAction
+import controllers.actions.{DataRetrievalAction, MrnRequiredActionImpl}
```

Comment:
Uhm. I would not go with the implementation class, MrnRequiredActionImpl, but with the trait.
Check the 2 comments below

Reply:
Fair enough, I had just looked at how it was done in `HowManyFilesUploadControllerSpec`, I've updated it now.